### PR TITLE
Added tests for certificate chain validation using GetCertificateChain command

### PR DESCRIPTION
**Pull request by hpya93**
Hi @jhand2 
Could you please review this? I took latest code and built the simulator afresh. The OIDs came as per spec.
- Have added test support for GetCertificateChain command
- Have added cert chain validation for intermediate CA certs collected from GetCertificateChain using x509.Verify() 
- Have added DPE leaf cert chain validation by calling GetCertificateChain

**Query 01**
- The cert chain validation fails if there any UnhandledCriticalExtension that crypto/x509  package doesnt know.
- It throws UnhandledCriticalExtensionError
- In code documentation they have suggested to overwrite that field with empty array if we know those extensions prior based on this comment [](https://cs.opensource.google/go/go/+/refs/tags/go1.21.0:src/crypto/x509/x509.go;l=714) 
- **Is this correct?**

**Query 02**
- Now I get "UnknownAuthorityError" on cert chain validation for leaf. It is because the SubjectName of cert I collect from GetCertificateChain Command doesnot match the IssuerName field of CertifyKey DPE leaf cert.
**- Can you please provide valid cert chain in response instead of dummy certs to validate positive cases?**

**Query 03**
- I see that the DPE leaf cert still returns errors in linting. Can we have a sample certificate from CertifyKey response without lint errors?
```
"e_ext_authority_key_identifier_missing":{
      "Status":Error (6),
      "Details":""
   } "e_ext_authority_key_identifier_no_key_identifier":{
      "Status":Error (6),
      "Details":""
   }"e_key_usage_incorrect_length":{
      "Status":Error (6),
      "Details":"the key usage ([128 1]) contains a value that is out of bounds of the range of possible KU values. (raw ASN: [3 3 0 128 1])"
   }"e_wrong_time_format_pre2050":{
      "Status":Error (6),
      "Details":""
   }

```

### CODE REVIEW

**Comment by jhand2** [comment]
> **Query 01**
> 
> * The cert chain validation fails if there any UnhandledCriticalExtension that crypto/x509  package doesnt know.
> * It throws UnhandledCriticalExtensionError
> * In code documentation they have suggested to overwrite that field with empty array if we know those extensions prior based on this comment
> * **Is this correct?**

We should first check that we know all of the OIDs for the unhandled extensions and make sure they are expected to be there. And then yes, overwriting with an empty array once they are known is the right thing to do.

> 
> **Query 02**
> 
> * Now I get "UnknownAuthorityError" on cert chain validation for leaf. It is because the SubjectName of cert I collect from GetCertificateChain Command doesnot match the IssuerName field of CertifyKey DPE leaf cert.
>   **- Can you please provide valid cert chain in response instead of dummy certs to validate positive cases?**

Thanks for pointing this out. We will fix this. I've opened this issue to track: https://github.com/chipsalliance/caliptra-dpe/issues/172

The caliptra emulator target has a valid cert chain once https://github.com/chipsalliance/caliptra-sw/pull/717 is merged. So that may be a good option for testing (once the task to hook up the emulator to these tests is done)

> 
> **Query 03**
> 
> * I see that the DPE leaf cert still returns errors in linting. Can we have a sample certificate from CertifyKey response without lint errors?
> 

This is being tracked in https://github.com/chipsalliance/caliptra-dpe/issues/74. We will work to address this, but it will not be right away. You can ignore the lint errors for now

File: verification/getCertificateChain_test.go [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+
+	getCertificateChainResp, err := client.GetCertificateChain(&getCertificateChainReq)
+	if err != nil {
+		t.Fatalf("Could not get Certificate Chain: %v", err)
+	}
+
+	//fmt.Println(getCertificateChainResp)
+	checkCertificateChain(t, getCertificateChainResp.CertificateChain)
+
+	// TODO: When DeriveChild is implemented, call it here to add more TCIs and call GetCertificateChain again.
```
Comment (jhand2) [inline verification/getCertificateChain_test.go:73]: This isn't relevant to GetCertificateChain in the same way it is for CertifyKey

File: verification/client.go [hunk 2]
```diff
@@ -192,6 +197,41 @@ func (c *Client[CurveParameter, Digest]) CertifyKey(cmd *CertifyKeyReq[Digest])
 	}, nil
 }
 
+// GetCertificateChain calls the DPE GetCertificateChain command.
+func (c *Client[_, _]) GetCertificateChain(cmd *GetCertificateChainReq) (*GetCertificateChainResp, error) {
```
Comment (jhand2) [inline verification/client.go:201]: The goal of this function is just to get the full cert chain, so I don't think you need to pass in GetCertificateChainReq. Instead this function can internally start from offset 0 and read until it gets to the end.

**Review by jhand2 — COMMENTED** [review]
Overall this looks good. Left a couple suggestions.

File: verification/getCertificateChain_test.go (hunk 1 above)
Comment (hpya93) [reply verification/getCertificateChain_test.go]: Did you mean the comment? Yes I removed it.

File: verification/client.go (hunk 2 above)
Comment (hpya93) [reply verification/client.go]: Yes, moved it into the client function

**Comment by hpya93** [comment]
@jhand2 
Hi, I have raised a PR based on your comments, can you please review and merge if ok? 
Thanks!

**Pull request closed without merging**